docs/tutorial.rst

```diff
@@ -1,149 +1,397 @@
-If you don't know what to enter, press Enter to stick with the defaults.
+    cd <project_slug>
+    mamba env create -f tests/CI_docker/context/environment_<project_slug>.yml
+    mamba activate <project_slug>
+    pip install .
+    make pytest
+    make lint
+    make urlcheck
+    make docs
 
+which are respectively for testing the whole package, lint style, urls, and documentation.
 
-Step 3: Create a GitHub Repo
+4. Prepare Gitlab repository
 ----------------------------
 
-Go to your GitHub account and create a new repo named ``mypackage``, where ``mypackage`` matches the ``[project_slug]`` from your answers to running cookiecutter. This is so that pyup.io can find it when we get to Step 5.
+After making sure that project has been successfully created, we need to create a blank gitlab repository and push the generated directory structure to it. To achieve this, follow these steps:
+
+4.1 Create repository using Web UI
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+On gitlab website, under the sub-group you want:
+
+* Create a new project by clicking on **New project**
+* Click on **Create blank project**
+* Give a **project name** at your choice
+* For the **project slug** pick the same name as the one given once you created directory at step 7 (check table above)
+* It's very important to unset the option ``Initialize repository with a README``
+* Click on **Create project**
+
+4.2. Push the repository
+~~~~~~~~~~~~~~~~~~~~~~~~
 
-You will find one folder named after the ``[project_slug]``. Move into this folder, and then setup git to use your GitHub repo and upload the code:
+Once you created the blank repository, follow the instructions to push the directory to the remote repository.
+
+They are summarized here:
 
 .. code-block:: bash
 
-    cd mypackage
-    git init .
+    cd <project_slug>
+    git init --initial-branch=main
+    git remote add origin git@git.gfz-potsdam.de:<group/subgroup/project_slug>.git
     git add .
-    git commit -m "Initial skeleton."
-    git remote add origin git@github.com:myusername/mypackage.git
+    git commit -m "Initial commit"
     git push -u origin main
 
-Where ``myusername`` and ``mypackage`` are adjusted for your username and package name.
+Now on you browser, refresh the page to see the repository.
 
-You'll need a ssh key to push the repo. You can `Generate`_ a key or `Add`_ an existing one.
+In the above code, the group is our directory in the gitlab (here is fernlab) and the subgroup is the text comes between the group and the project_name name.
 
-.. _`Generate`: https://help.github.com/articles/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent/
-.. _`Add`: https://help.github.com/articles/adding-a-new-ssh-key-to-your-github-account/
+5. Create and attach a runner to the repository
+-----------------------------------------------
 
+Login to a mefe machine you like to deploy the runner using your gfz email id (i.e.  ``<email-id>@gfz-potsdam.de``) and your master password:
```

Comment:
In this chapter we should not use the names mefe and the path to our fernlab directory but generic examples like in Step [5. Create and attach a runner to the repository.](https://github.com/FernLab/cookiecutter-py-package/blob/main/docs/tutorial.rst#step-5-create-and-attach-a-runner-to-the-repository)

Reply:
OK! Can you check to see if I resolved it?